.github/workflows/publish.yaml

```diff
@@ -0,0 +1,133 @@
+name: FUT Infrastructure Implementation Guide publish pipeline
+
+on:
+  workflow_dispatch: # Enable manual run
+  push:
+    branches: [ master ]
+    tags: ['*']
+  pull_request:
+    branches: [ master ]
+
+env:
+  # https://confluence.hl7.org/display/FHIR/IG+Publisher+CLI
+  ig_cmd: java -Xmx4g -jar /input-cache/publisher.jar
+
+jobs:
+  # Check if the IG transpiles without errors
+  check-transpile:
+    name: Transpile IG
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/fut-infrastructure/ig-publisher/ig-publisher:latest
+    steps:
+      # Checkout the repos needed
+      - uses: actions/checkout@v4
+
+        # https://confluence.hl7.org/display/FHIR/IG+Publisher+CLI
+      - name: ig transpile
+        run: $ig_cmd -ig .
+
+  # Transpile and publish the IG, if it was pushed to main with a tag
+  release:
+    name: Transpile and publish to the IG release
+    # A release is defined as a tag pushed to main
+    if: ${{ github.event_name == 'push' && github.ref_type == 'tag' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/fut-infrastructure/ig-publisher/ig-publisher:latest
```

Comment:
`latest` is a bit dangerous - can we pin it?

Reply:
We could, but it's kind of pinned by creating new versions with tags in https://github.com/fut-infrastructure/ig-publisher/blob/main/.github/workflows/container-build.yaml

We always want to use the newest publisher version. We can always temporarily pin it to an old one if something breaks.

Reply:
I agree with @kdhtrifork but it seems contained for now.